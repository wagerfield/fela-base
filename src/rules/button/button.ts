export interface ButtonRuleOptions {
  background?: any
  userSelect?: any
  boxSizing?: any
  outline?: any
  padding?: any
  border?: any
  cursor?: any
}

export default ({
  background = "transparent",
  userSelect = "none",
  boxSizing = "border-box",
  outline = "none",
  padding = 0,
  border = "none",
  cursor = "pointer"
}: ButtonRuleOptions = {}) => ({
  backgroundColor: background,
  userSelect,
  boxSizing,
  outline,
  padding,
  border,
  cursor
})
