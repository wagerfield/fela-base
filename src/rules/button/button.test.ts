import button from "./button"

describe("button(props)", () => {
  it("renders expected snapshots", () => {
    expect(button()).toMatchSnapshot()
    expect(
      button({
        background: "red"
      })
    ).toMatchSnapshot()
  })
})
