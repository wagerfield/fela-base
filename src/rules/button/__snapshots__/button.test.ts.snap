// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button(props) renders expected snapshots 1`] = `
Object {
  "backgroundColor": "transparent",
  "border": "none",
  "boxSizing": "border-box",
  "cursor": "pointer",
  "outline": "none",
  "padding": 0,
  "userSelect": "none",
}
`;

exports[`button(props) renders expected snapshots 2`] = `
Object {
  "backgroundColor": "red",
  "border": "none",
  "boxSizing": "border-box",
  "cursor": "pointer",
  "outline": "none",
  "padding": 0,
  "userSelect": "none",
}
`;
